sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -0,0 +1,503 @@
+            scope.Start();
+            try
+            {
+                var stream = SerializeManifest(manifest);
+                string tagOrDigest = options.Tag ?? OciBlobDescriptor.ComputeDigest(stream);
+                ResponseWithHeaders<ContainerRegistryCreateManifestHeaders> response = _restClient.CreateManifest(_repositoryName, tagOrDigest, manifest, ManifestMediaType.OciManifest.ToString(), cancellationToken);
+                return Response.FromValue(new UploadManifestResult(response.Headers.DockerContentDigest), response.GetRawResponse());
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Upload a manifest for an OCI Artifact.
+        /// </summary>
+        /// <param name="manifest">The manifest to upload.</param>
+        /// <param name="options">Options for configuring the upload operation.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public async virtual Task<Response<UploadManifestResult>> UploadManifestAsync(OciManifest manifest, UploadManifestOptions options = default, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(manifest, "manifest");
+
+            options ??= new UploadManifestOptions();
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRegistryBlobClient)}.{nameof(UploadManifest)}");
+            scope.Start();
+            try
+            {
+                var stream = SerializeManifest(manifest);
+                string tagOrDigest = options.Tag ?? OciBlobDescriptor.ComputeDigest(stream);
+                ResponseWithHeaders<ContainerRegistryCreateManifestHeaders> response = await _restClient.CreateManifestAsync(_repositoryName, tagOrDigest, manifest, ManifestMediaType.OciManifest.ToString(), cancellationToken).ConfigureAwait(false);
+                return Response.FromValue(new UploadManifestResult(response.Headers.DockerContentDigest), response.GetRawResponse());
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private static Stream SerializeManifest(OciManifest manifest)
+        {
+            MemoryStream stream = new MemoryStream();
+            Utf8JsonWriter jsonWriter = new Utf8JsonWriter(stream);
+            ((IUtf8JsonSerializable)manifest).Write(jsonWriter);
+            jsonWriter.Flush();
+            return stream;
+        }
+
+        private static OciManifest DeserializeManifest(Stream stream)
+        {
+            using var document = JsonDocument.Parse(stream);
+            return OciManifest.DeserializeOciManifest(document.RootElement);
+        }
```

Comment:
Jut curious why this isn't generated

Reply:
There is a bug in the service where upload/download can fail when done directly from a stream instead of the serialized type: https://msazure.visualstudio.com/AzureContainerRegistry/_workitems/edit/10780642 

This is a workaround for that until the bug is fixed.